includes/Core/Authentication/Google_Proxy.php

```diff
@@ -399,24 +450,59 @@ public function unregister_site( Credentials $credentials ) {
 		return $this->request( self::OAUTH2_DELETE_SITE_URI, $credentials );
 	}
 
+	/**
+	 * Registers the site on the proxy.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string|WP_Error Redirect URL on success, otherwise an error.
+	 */
+	public function register_site() {
+		return $this->sync_site_fields( null, null );
+	}
+
 	/**
 	 * Synchronizes site fields with the proxy.
 	 *
 	 * @since 1.5.0
+	 * @since n.e.x.t Updated the function to return redirect URL.
 	 *
 	 * @param Credentials $credentials Credentials instance.
 	 * @param string      $mode        Sync mode.
-	 * @return array|WP_Error Response of the wp_remote_post request.
+	 * @return string|WP_Error Redirect URL on success, otherwise an error.
 	 */
-	public function sync_site_fields( Credentials $credentials, $mode = 'async' ) {
-		return $this->request(
+	public function sync_site_fields( Credentials $credentials = null, $mode = 'async' ) {
+		$response = $this->request(
 			self::OAUTH2_SITE_URI,
 			$credentials,
 			array(
-				'mode' => $mode,
-				'body' => $this->get_site_fields(),
+				'return' => 'response',
+				'mode'   => $mode,
+				'body'   => array_merge(
+					$this->get_site_fields(),
+					$this->get_user_fields(),
+					$this->get_metadata_fields(),
+					array(
+						'scopes' => implode( ' ', $this->required_scopes ),
```

Comment:
Similar to OAuth, this parameter has to be called `scope` (singular), even though it's multiple scopes.

Reply:
Updated.